File: Prácticas/P4/RUBY/Civitas/lib/sorpresa_ir_carcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "tablero.rb"
require_relative "diario.rb"
require_relative "jugador.rb"
require_relative "mazo_sorpresas.rb"

module Civitas
  class SorpresaIrCarcel < Sorpresa
    def initialize(tablero,texto)
      @tablero = tablero
      @texto = texto
    end
    
    def aplicar_a_jugador(actual,todos)
      if(jugador_correcto(actual,todos))
        informe(actual,todos)
        todos[actual].encarcelar(@tablero.get_carcel)
      end
    end
    
    def to_s
      to_s="Sorpresa ir a carcel {#{@texto}, valor=#{@valor}}"
    end
  end
end

File: Prácticas/P4/JAVA/Civitas/src/civitas/SorpresaIrCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Jose Padial
 */
public class SorpresaIrCarcel extends Sorpresa{
    private Tablero tablero;
    
    SorpresaIrCarcel(Tablero tablero){
        this.tablero = tablero;
    }
    
    @Override
    void aplicarJugador(int actual, ArrayList<Jugador> todos){
        if(todos.get(actual) != null){
            informe(actual,todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
    }

    @Override
    public String toString() {
        return "SorpresaIrCarcel{" + "tablero=" + tablero + '}';
    }
}
